userguide/tutorials/PostgreSQL.adoc

```diff
@@ -0,0 +1,78 @@
+. Create a user corresponding to Kill Bill. You can run the following command :
+[source,sql]
+ CREATE ROLE killbilluser WITH LOGIN INHERIT CREATEDB CREATEROLE NOREPLICATION PASSWORD 'killbill';
+
+ . Create a database corresponding to Kill Bill. You can run the following command:
+[source,sql]
+ CREATE DATABASE killbill WITH OWNER = killbilluser;
+
+
+. Connect to the Kill Bill database created above. In psql, you can run the following:
+[source,sql]
+  \connect killbill;
+
+. Create the Kill Bill schema. Run the following command:
+[source,sql]
+  CREATE SCHEMA killbillschema authorization killbilluser;
+
+. Connect to the schema created above using the following command:
+[source,sql]
+  set schema 'killbillschema';
+
+. Execute https://github.com/killbill/killbill/blob/master/util/src/main/resources/org/killbill/billing/util/ddl-postgresql.sql[this] schema extension DDL.
+
+. Run the Kill Bill DDL https://docs.killbill.io/latest/ddl.sql[here]. In psql, you  can run the following command:
+[source,sql]
+  \i <ddl_file_path>
+
+
+. Grant privileges to the schema using the following command:
+[source,sql]
+  GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA killbillschema TO killbilluser;
+
+=== PostgreSQL kaui Configuration
+
+In order to configure PostgreSQL, you need to follow the steps given below (These steps can be executed either via the command line *psql* tool or via *pgAdmin*):
+
+.  Create a database corresponding to kaui. You can run the following command:
+[source,sql]
+ CREATE DATABASE kaui WITH OWNER = killbilluser;
+
+.  Connect to the kaui database created above. In psql, you can run the following:
+[source,sql]
+  \connect kaui;
+
+. Create the Kill Bill schema. Run the following command:
+[source,sql]
+  CREATE SCHEMA killbillschema authorization killbilluser;
+
+. Connect to the schema created above using the following command:
+[source,sql]
+  set schema 'killbillschema';
+
+.  Execute https://github.com/killbill/killbill/blob/master/util/src/main/resources/org/killbill/billing/util/ddl-postgresql.sql[this] schema extension DDL.
+
+.  Run the Kaui DDL https://github.com/killbill/killbill-admin-ui/blob/master/db/ddl.sql[here].
+
+.  Grant privileges to the schema using the following command:
+[source,sql]
+  GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA killbillschema TO killbilluser;
+  GRANT ALL PRIVILEGES ON ALL SEQUENCES IN SCHEMA killbillschema TO killbilluser;
+
+.  Modify the `PROJECT_ROOT/profiles/killbill/src/main/resources/killbill-server.properties` OR `TOMCAT_HOME/conf/catalina.properties` as the case may be. Add the following properties :
+[source,properties]
+ #KAUI postgresql section
+  kaui.db.url=jdbc:postgresql://127.0.0.1:5432/kaui
+  kaui.db.username=killbilluser
+  kaui.db.password=killbill
+
+.  Modify the `TOMCAT_HOME/webapps2/ROOT/WEB-INF/config/database.yml` adapter on line 3 as :
+[source,properties]
+  <% adapter = java.lang.System.getProperty('kaui.db.adapter', ENV['KAUI_DB_ADAPTER'] || 'postgresql') %>
```

Comment:
Instead of modifying the line, can't the user simply set `KAUI_DB_ADAPTER= postgresql ` (e.g. in Tomcat setenv.sh)?

The default line is, so the environment variable `KAUI_DB_ADAPTER` should be read by default:

```
<% adapter = java.lang.System.getProperty('kaui.db.adapter', ENV['KAUI_DB_ADAPTER'] || 'mysql2') %>
```

Reply:
@pierre, do you mean [this](https://github.com/killbill/killbill-docs/pull/370#discussion_r895414417) ?

Reply:
Ah, yes, that works too and no need for the user to tweak the yaml file (which could be a bit error prone). Sorry I missed it, too many GitHub notifications 🙈